artifactory/manager.go

```diff
@@ -272,10 +272,10 @@ func (sm *ArtifactoryServicesManager) GetReplication(repoKey string) ([]utils.Re
 	return getPushReplicationService.GetReplication(repoKey)
 }
 
-func (sm *ArtifactoryServicesManager) GetArtifactoryVersion() (string, error) {
-	versionService := services.NewVersionService(sm.client)
-	versionService.ArtDetails = sm.config.GetServiceDetails()
-	return versionService.GetArtifactoryVersion()
+func (sm *ArtifactoryServicesManager) CreateSystemService() *services.SystemService {
```

Comment:
It looks like this function as asymmetric compared to the other functions in this file.
Notice that non of the other functions return a service. This is the only function which returns a service. I suggest replacing it with two functions - GetVersion and GetServiceId (both functions will use the new SystemService of course).